packages/valory/skills/task_submission_abci/behaviours.py

```diff
@@ -1261,8 +1271,9 @@ def _num_to_bytes(value: int) -> bytes:
                     ) = yield from self._get_encoded_deliver_data(
                         request_ids, deliver_datas
                     )
-                    request_ids = final_request_ids
-                    deliver_datas = final_datas
+                    if final_request_ids and final_datas:
+                        request_ids = final_request_ids
+                        deliver_datas = final_datas
```

Comment:
The nesting here is extreme. Please try to avoid such nesting as in this method. For example, first lines of the method:

```python
if len(marketplace_done_tasks) <= 0:
    return []
```

This is already one level less and easier to read.

You can skip this comment if you'd like, as the codebase follows a similar pattern and this won't have a significant impact. However, it's something to keep in mind as a general best practice.